### DISCUSSION TITLE
Strange behavior with modals

### DISCUSSION THREAD

**Discussion by Just-Dominic** [primefaces/community#1072] [PrimeFaces]
Hi,

I have a widget (hotjar) dat uses a textarea for users to leave their feedback. It works great throughout the whole application. Except when I open a dialog using the attribute modal=true . The pointer-events in the widget are working properly and the widget is stacked above the dialog as intended. The only thing that is not working is keyboard input inside the widget.

It looks like the attribute modal=true is disabling all input fields in the DOM besides the ones in the active dialog. The widget resides on body level so I've tried to append the widget to the dialog when it gets opened with jQuery. No difference..

Do you have any ideas cause I have none...

![image](https://github.com/primefaces/community/assets/144908012/8def4594-a63e-4811-acb4-8484c91fccd0)

**Comment by melloware** [comment]
Please provide an executable example using the [PrimeFaces Test](https://github.com/primefaces/primefaces-test) project.  It is the only way developers can debug your problem to help.

I will say yes the only components that recieve keyboard input are ones attached inside the dialog.  That is the whole point of `modal`.  Also have you looked at PFE Slideout for taking user Feedback: https://www.primefaces.org/showcase-ext/sections/slideout/basicUsage.jsf

That is what we use at my client. See this feedback example: https://www.primefaces.org/showcase-ext/sections/slideout/advancedUsage.jsf

**Comment by Just-Dominic** [comment]
Hi, thanks for the reply. 

We work with sensitive data so I can't provide a test environment. I was hoping you ran into this issue before.

Thanks for your suggestion but we will stick to hotjar for now. I need extensive user data.

I am thinking about adding an exclude somehow to the modal...

**Reply by melloware** [reply]
Let me know what you come up with.  But the code if you want to look at it or debug it for keeping your focus inside the modal dialog is right here: https://github.com/primefaces/primefaces/blob/030ae7d32d4ffd76edd8c01564a410464f62eac5/primefaces/src/main/resources/META-INF/resources/primefaces/core/core.utils.js#L158-L222

particularly focus on this zIndex part: 

```js
else if (!target.is(document.body) && (!$documentInIframe && target.zIndex() < zIndex && target.parent().zIndex() < zIndex)) {
                    event.preventDefault();
                }
```
